### ISSUE TITLE
pod presets getting rid of extra information

### ISSUE THREAD

**Issue by karmab** [kubevirt/kubevirt#791]
## Description
i f i create a vm from a yaml file containing extra information, this extra information gets deleted while if i do editing it on an existing vm, it stays

## Expected
the extra information at vm's creation is kept

```
metadata:
  name: testvm-registrydisk
apiVersion: kubevirt.io/v1alpha1
kind: VirtualMachine
myinfo: bigsecret
spec:
  domain:
    resources:
      requests:
        memory: 64M
    devices:
      disks:
      - name: registrydisk
        volumeName: registryvolume
        disk: {}
  volumes:
    - name: registryvolume
      registryDisk:
        image: vmdisks/fedora25:latest
```

**Comment by rmohr** [comment]
Since there is no schema validation enforced at the moment, if no controller updates the vm, this extra information will be accepted by the system and stays there. However, if a controller manipulates the vm, it will only recognise fields which are part of the api, and everything which is not part of the api will be removed on the update.

In your case, the `myinfo: bigsecret` is put on the document root. My suspect that you wanted to add an annotation or a label. The root paths for them are `metadata.annotations` and `metadata.labels`. These will stay.

Related to #556. Once #556 is done, unknown fields will either not be persisted on a POST/PUT, or the object may not even be accepted.

**Comment by fabiand** [comment]
Just to make it clearer what Roman said: You changed the object, and did not add a label.
From the IRC conversation I learned that you wanted to add labels, but labels belong into the metadata section as follows:

```
kind: VirtualMachine:
metadata:
  name: foobar
  labels:
    karimb.io/foo: bar
spec:
  …
```

**Comment by karmab** [comment]
will use annotations
